gitlab/__init__.py

```diff
@@ -143,33 +137,31 @@ def _format_string(string):
             return quote_plus(string)
         return string
 
-    @staticmethod
-    def success_or_raise(response, status_codes, default_response=None):
+    def success_or_raise(self, response, default_response=None):
         """
         Check if request was successful or raises an HttpError
 
         :param response: Response Object to check
-        :param status_codes: List of Ints, Valid status codes to check for
         :param default_response: Return value if JSONDecodeError
-        :return: Dictionary containing response data
-        :raise: HttpError: If invalid response returned
+        :returns dict: Dictionary containing response data
+        :returns bool: :obj:`False` on failure when exceptions are suppressed
+        :raises requests.exceptions.HTTPError: If invalid response returned
         """
-        if default_response is None:
-            default_response = {}
+        response_json = default_response
```

Comment:
I didn't think of this when I was creating this method. I like your pattern here.